plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/HiveMetastoreModule.java

```diff
@@ -42,9 +43,8 @@ protected void setup(Binder binder)
     {
         if (metastore.isPresent()) {
             binder.bind(HiveMetastoreFactory.class).annotatedWith(RawHiveMetastoreFactory.class).toInstance(HiveMetastoreFactory.ofInstance(metastore.get()));
-            MetastoreTypeConfig metastoreTypeConfig = new MetastoreTypeConfig();
-            metastoreTypeConfig.setMetastoreType("provided");
-            binder.bind(MetastoreTypeConfig.class).toInstance(metastoreTypeConfig);
+            configBinder(binder).bindConfigDefaults(MetastoreTypeConfig.class, config -> config.setMetastoreType("provided"));
+            configBinder(binder).bindConfig(MetastoreTypeConfig.class);
```

Comment:
It is wrong. Here the code intention is _not_ to expose any configurable properties

Reply:
I talked with @homar about this and was under the impression that this was done this way because there was no better way. I kinda understand why it's like this, but the current way is wrong too :)

> Here the code intention is not to expose any configurable properties

I get that. But this code path is only used when `HiveQueryRunner` is used, so I added an assertion there to prevent setting this property. I was hoping this would have the same effect 🤷